spark3/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.procedures;
+
+import java.util.function.Function;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.spark.source.SparkTable;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.catalyst.parser.ParseException;
+import org.apache.spark.sql.catalyst.parser.ParserInterface;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.Procedure;
+import org.apache.spark.sql.execution.CacheManager;
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation;
+import scala.Option;
+import scala.collection.Seq;
+
+abstract class BaseProcedure implements Procedure {
+  private final SparkSession spark;
+  private final TableCatalog catalog;
+
+  protected BaseProcedure(TableCatalog catalog) {
+    this.spark = SparkSession.active();
+    this.catalog = catalog;
+  }
+
+  protected <T> T modifyIcebergTable(String namespace, String tableName, Function<org.apache.iceberg.Table, T> func) {
+    Preconditions.checkArgument(!namespace.isEmpty(), "Namespace cannot be empty");
+    Preconditions.checkArgument(!tableName.isEmpty(), "Table name cannot be empty");
+
+    Identifier ident = toIdentifier(namespace, tableName);
+    SparkTable sparkTable = loadSparkTable(ident);
+    org.apache.iceberg.Table icebergTable = sparkTable.table();
+
+    T result = func.apply(icebergTable);
+
+    refreshSparkCache(ident, sparkTable);
+
+    return result;
+  }
+
+  // we have to parse both namespace and name as they may be quoted
+  protected Identifier toIdentifier(String namespaceAsString, String name) {
```

Comment:
This place requires discussion. Right now, I allow calling procedures like this: 
```
CALL cat.system.rollback_to_snapshot_id('`name.space`', '`table.name`', snapshot_id)
```

Where the namespace and table names are strings but they may be quoted.

Reply:
If they are strings, then they should be passed using string escapes. I would expect back-tick to be preserved.

Reply:
Thinking about this a bit more, maybe this is okay. I can't think of another way to quote name.space if it is an identifier.

Reply:
Will we be able to construct a valid namespace then? I think the tricky part is that `.` is not a special character but may indicate a namespace boundary.

Reply:
I'd love to find a better approach, though.

Reply:
This is fine for now.